pyvista/core/filters/uniform_grid.py

```diff
@@ -414,3 +414,265 @@ def image_threshold(
         # run the algorithm
         _update_alg(alg, progress_bar, 'Performing Image Thresholding')
         return _get_output(alg)
+
+    def fft(self, output_scalars_name='ImageScalars', progress_bar=False):
+        """Apply the fast fourier transform to the active scalars.
+
+        The input can have real or complex data in any components and data
+        types, but the output is always complex doubles, with the first
+        component containing real values and the second component containing
+        imaginary values. The filter is fastest for images that have power of
+        two sizes.
+
+        The filter is fastest for images that have power of two sizes. The
+        filter uses a butterfly diagram for each prime factor of the
+        dimension. This makes images with prime number dimensions (i.e. 17x17)
+        much slower to compute. Multi dimensional (i.e volumes) FFT's are
+        decomposed so that each axis executes serially.
+
+        Parameters
+        ----------
+        output_scalars_name : str, optional
+            The name of the output scalars. By default this is
+            ``'ImageScalars'``.
+
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.UniformGrid
+            UniformGrid with applied FFT.
+
+        Examples
+        --------
+        Apply FFT to an example image.
+
+        >>> from pyvista import examples
+        >>> image = examples.download_moonlanding_image()
+        >>> fft_image = image.fft()
+        >>> fft_image.point_data  # doctest:+SKIP
+        pyvista DataSetAttributes
+        Association     : POINT
+        Active Scalars  : PNGImage
+        Active Vectors  : None
+        Active Texture  : None
+        Active Normals  : None
+        Contains arrays :
+        PNGImage                float64  (298620, 2)          SCALARS
+
+        See :ref:`image_fft_example` for a full example using this filter.
+
+        """
+        # check for active scalars, otherwise risk of segfault
+        if self.point_data.active_scalars_name is None:
+            raise ValueError('FFT filter requires active point scalars')
```

Comment:
In 47885591e3ed21d5ddacc942dfb527b0798ccfde, I implemented this. It's a bit of a pain as I could not reuse `pyvista.helpers.set_default_active_scalars` because that function sets point or cell scalars to be active. Logic could have been changed to enable setting a preference, but in some cases it doesn't make sense to be 100% DRY.